contract_sale_invoicing/models/contract.py

```diff
@@ -28,15 +28,10 @@ def _recurring_create_invoice(self, date_ref=False):
                         contract.partner_id.commercial_partner_id.ids,
                     ),
                     ("invoice_status", "=", "to invoice"),
-                    (
-                        "date_order",
-                        "<=",
-                        "{} 23:59:59".format(contract.recurring_next_date),
-                    ),
+                    ("date_order", ">=", contract.recurring_next_date),
```

Comment:
But that can't be correct. This method won't invoice past sale orders now. You can't change the entire behavior because a unit test fails. The issue must be somewhere else.

Reply:
@dsolanki-initos Could you check this comment and find why this change is needed as it is the contrary of the previous code ?